x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/AnalyticsCollectionService.java

```diff
@@ -74,20 +81,33 @@ public void putAnalyticsCollection(
         // This operation is supposed to be executed on the master node only.
         assert (state.nodes().isLocalNodeElectedMaster());
 
-        try {
-            analyticsCollectionResolver.collection(state, request.getName());
-            listener.onFailure(new ResourceAlreadyExistsException("analytics collection {} already exists", request.getName()));
-        } catch (ResourceNotFoundException e) {
-            AnalyticsCollection collection = new AnalyticsCollection(request.getName());
-            CreateDataStreamAction.Request createDataStreamRequest = new CreateDataStreamAction.Request(collection.getEventDataStream());
+        AnalyticsCollection collection = new AnalyticsCollection(request.getName());
+        CreateDataStreamAction.Request createDataStreamRequest = new CreateDataStreamAction.Request(collection.getEventDataStream());
+
+        ActionListener<AcknowledgedResponse> createDataStreamListener = ActionListener.wrap(
+            r -> listener.onResponse(new PutAnalyticsCollectionAction.Response(r.isAcknowledged(), request.getName())),
+            (Exception e) -> {
+                if (e instanceof ResourceAlreadyExistsException) {
+                    listener.onFailure(
+                        new ResourceAlreadyExistsException("analytics collection [{}] already exists", request.getName(), e)
+                    );
+                    return;
+                }
+
+                RestStatus status = RestStatus.INTERNAL_SERVER_ERROR;
+
+                if (e instanceof ElasticsearchException) {
+                    status = ((ElasticsearchException) e).status();
+                }
+
+                e = new ElasticsearchStatusException("error while creating analytics collection [{}]", status, e, request.getName());
```

Comment:
I _think_ status is already obtained from a wrapped exception when using ElasticsearchException, see [here](https://github.com/elastic/elasticsearch/blob/db0500ee24c9fed5de6e96eaf626b0d974251998/server/src/main/java/org/elasticsearch/ElasticsearchException.java#L232). So probably we just wrap the exception into a new exception with the needed message.

Reply:
The unwrapping does not work but I was able to do this instead of having a conditional:

```
e = new ElasticsearchStatusException(
  "error while creating analytics collection [{}]",
  ExceptionsHelper.status(e),
  e,
  request.getName()
);
```